File: exercism/go/bird-watcher/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	totalCount := 0

	for _, value := range birdsPerDay {
		totalCount += value
	}

	return totalCount

}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	totalCount := 0
	totalDays := week * 7

	for day := totalDays - 7; day < totalDays; day++ {
		totalCount += birdsPerDay[day]
	}

	return totalCount
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for day := 0; day < len(birdsPerDay); day += 2 {
		birdsPerDay[day] = birdsPerDay[day] + 1
	}

	return birdsPerDay
}

File: exercism/java/bird-watcher/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
      return this.birdsPerDay;
    }

    public int getToday() {
      return this.birdsPerDay[this.birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
      int lastIndex = this.birdsPerDay.length - 1;
      this.birdsPerDay[lastIndex] = this.birdsPerDay[lastIndex] + 1;
    }

    public boolean hasDayWithoutBirds() {
      for (int count: this.birdsPerDay) {
        if (count == 0) {
          return true;
        }
      }

      return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
      int total = 0;
      int totalDays = this.birdsPerDay.length;
      
      for (int i = 0; i < numberOfDays; i++) {
        if (i < this.birdsPerDay.length) {
          total += this.birdsPerDay[i];       
        }
      }

      return total;
    }

    public int getBusyDays() {
      int count = 0;
      
      for (int c: this.birdsPerDay) {
        if (c >= 5) {
          count++;
        }
      }

      return count;
    }
}
